firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionLifecycleService.kt

```diff
@@ -110,17 +111,18 @@ internal class SessionLifecycleService() : Service() {
      * given [Message]. This will determine if the foregrounding should result in the creation of a
      * new session.
      */
-    private fun handleForegrounding(msg: Message) {
-      Log.d(TAG, "Activity foregrounding at ${msg.getWhen()}")
-      if (!hasForegrounded) {
-        Log.d(TAG, "Cold start detected.")
-        hasForegrounded = true
-        newSession()
-      } else if (isSessionRestart(msg.getWhen())) {
-        Log.d(TAG, "Session too long in background. Creating new session.")
-        newSession()
+    private fun handleForegrounding(msg: Message) =
+      CoroutineScope(Dispatchers.instance.backgroundDispatcher).launch {
```

Comment:
Let's not do this on a different thread. The service guarantees thread safety by performing all the logic in the looper thread, so by moving this to the background we'd have to worry about concurrent modifications of member variables by different threads.

Reply:
Done